File: 219-contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> appearances = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (appearances.containsKey(nums[i])) {
                if ((i - appearances.get(nums[i]) <= k)) {
                    return true;
                }
            }
            appearances.put(nums[i], i);
        }
        return false;
    }
}

File: 219-contains-duplicate-ii/contains-duplicate-ii.kt
class Solution {
    fun containsNearbyDuplicate(nums: IntArray, k: Int): Boolean {
        var hashMap: HashMap<Int, Int> = HashMap<Int, Int>()

        for (i in 0..nums.size -1) {
            if (hashMap.containsKey(nums[i])) {
                if (i - hashMap.get(nums[i])!! <= k) {
                    return true;
                }
            }
            hashMap.put(nums[i], i)
        }
        return false;
    }
}